semgrep-core/src/core/ast/Find_target.ml

```diff
@@ -0,0 +1,68 @@
+(*
+   Find and filter targets.
+*)
+
+open Common
+module Resp = Semgrep_core_response_t
+
+let exclude_files_in_skip_lists roots =
+  let paths, skipped_paths =
+    Skip_code.filter_files_if_skip_list ~root:roots roots
+  in
+  let skipped =
+    skipped_paths
+    |> List.map (fun path ->
+           {
+             Resp.path;
+             reason = Excluded_by_config;
+             details = "excluded by 'skip list' file";
+             skipped_rule = None;
+           })
+  in
+  (paths, skipped)
+
+(*
+   Some source files are really huge (> 20 MB) and they cause
+   some annoying 'out of memory' crash that sometimes even the use
+   of mem_limit_mb above does not solve.
+
+   We could configure the size limit based on a per-language basis if we
+   know that a language parser can handle larger files.
+*)
+let filter_by_size _lang paths =
+  let max_bytes = 5_000_000 in
```

Comment:
Should this be a CLI option?

Reply:
maybe if anyone asks for it. I'm not convinced it's a good hack at all and would rather keep it internal for now.

Reply:
I just created the CLI option.